service/matching/taskQueueManager.go

```diff
@@ -474,42 +478,42 @@ func (c *taskQueueManagerImpl) MutateVersioningData(ctx context.Context, mutator
 	if err != nil {
 		return err
 	}
+	if !c.taskQueueID.IsRoot() {
+		return nil
+	}
 	// If this is the root partition, notify partitions that they should fetch changed data from us.
-	if c.taskQueueID.IsRoot() {
-		numParts := util.Max(c.config.NumReadPartitions(), c.config.NumWritePartitions())
-		wg := &sync.WaitGroup{}
-		for i := 1; i < numParts; i++ {
-			wg.Add(1)
-			go func(i int) {
-				tq := c.taskQueueID.mkName(i)
-				_, err := c.matchingClient.InvalidateTaskQueueMetadata(ctx,
-					&matchingservice.InvalidateTaskQueueMetadataRequest{
-						NamespaceId:    c.taskQueueID.namespaceID.String(),
-						TaskQueue:      tq,
-						VersioningData: newDat,
-					})
-				if err != nil {
-					c.logger.Warn("Failed to notify sub-partition of invalidated versioning data",
-						tag.WorkflowTaskQueueName(tq), tag.Error(err))
-				}
-				wg.Done()
-			}(i)
-			wg.Wait()
-		}
+	numParts := util.Max(c.config.NumReadPartitions(), c.config.NumWritePartitions())
+	wg := &sync.WaitGroup{}
+	for i := 1; i < numParts; i++ {
+		wg.Add(1)
+		go func(i int) {
+			tq := c.taskQueueID.mkName(i)
+			_, err := c.matchingClient.InvalidateTaskQueueMetadata(ctx,
+				&matchingservice.InvalidateTaskQueueMetadataRequest{
+					NamespaceId:    c.taskQueueID.namespaceID.String(),
+					TaskQueue:      tq,
+					VersioningData: newDat,
+				})
+			if err != nil {
+				c.logger.Warn("Failed to notify sub-partition of invalidated versioning data",
+					tag.WorkflowTaskQueueName(tq), tag.Error(err))
+			}
+			wg.Done()
+		}(i)
 	}
+	wg.Wait()
 	return nil
 }
 
 func (c *taskQueueManagerImpl) InvalidateMetadata(request *matchingservice.InvalidateTaskQueueMetadataRequest) error {
 	if request.GetVersioningData() != nil {
-		if c.taskQueueID.IsRoot() {
+		if c.taskQueueID.IsRoot() && c.taskQueueID.taskType == enumspb.TASK_QUEUE_TYPE_WORKFLOW {
 			// Should never happen. Root partitions do not get their versioning data invalidated.
-			c.logger.Warn("A root partition was told to invalidate its versioning data, this should not happen",
-				tag.WorkflowTaskQueueName(c.taskQueueID.name))
+			c.logger.Warn("A root workflow partition was told to invalidate its versioning data, this should not happen")
 			return nil
```

Comment:
return an error?

Reply:
Eh, I dunno if it's worth it. It shouldn't happen, but it also makes sense to just ignore it and not do it, could go either way.